salt/metalk8s/kubernetes/apiserver-proxy/files/apiserver-proxy.conf.j2

```diff
@@ -11,20 +11,21 @@ events {
 }
 
 {%- if pillar.get('apiserver_ip') %}
-{%-     set masters_request = {'ip': pillar.apiserver_ip} %}
+{%-     set apiservers = [pillar.apiserver_ip] %}
 {%- else %}
-{%-     set masters_request = salt['mine.get'](
-            tgt='I@metalk8s:nodes:*:roles:master',
-            fun='control_plane_ip',
-            tgt_type='compound',
-        )
+{%-     set masters = salt['metalk8s.minions_by_role']('master') %}
+{%-     if not masters %}
+            {{ raise("No master nodes in pillar 'metalk8s:nodes', "
+                     ~ "can't deploy apiserver-proxy") }}
+{%-     else %}
+{%-         set master_ips = salt['mine.get'](
+                tgt=masters | join(','),
+                fun='control_plane_ip',
+                tgt_type='list',
+            )
 %}
-{%- endif %}
-{%- if not masters_request %}
-{# In this case, we're (likely) bootstrapping #}
-{%-   set apiservers = ['127.0.0.1'] %}
-{%- else %}
-{%-   set apiservers = masters_request.values() | sort %}
+{%-         set apiservers = master_ips.values() | sort %}
```

Comment:
You still need to do the bootstrap trick here since mine does not work without master (I think you can check that `grains.id` is in the `masters` list and add 127.0.0.1 in this case (only if mine empty of course) it will be PERFECT :smiling_face_with_three_hearts:)

Reply:
Ah, right :sweat_smile: